components/proxy/src/test/java/com/hotels/styx/startup/StyxServerComponentsTest.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.startup;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.hotels.styx.Environment;
+import com.hotels.styx.StyxConfig;
+import com.hotels.styx.api.HttpInterceptor.Chain;
+import com.hotels.styx.api.configuration.Configuration;
+import com.hotels.styx.api.metrics.codahale.CodaHaleMetricRegistry;
+import com.hotels.styx.api.plugins.spi.Plugin;
+import com.hotels.styx.api.service.spi.StyxService;
+import com.hotels.styx.proxy.plugin.NamedPlugin;
+import com.hotels.styx.startup.StyxServerComponents.LoggingSetUp;
+import com.hotels.styx.support.api.BlockingObservables;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.hotels.styx.api.HttpRequest.Builder.get;
+import static com.hotels.styx.api.HttpResponse.Builder.response;
+import static com.hotels.styx.proxy.plugin.NamedPlugin.namedPlugin;
+import static com.hotels.styx.support.matchers.IsOptional.isValue;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static rx.Observable.just;
+
+// TODO it is still difficult to decouple + test the configured plugins + services
+public class StyxServerComponentsTest {
+    @Test
+    public void setsUpLoggingOnBuild() {
+        LoggingSetUp loggingSetUp = mock(LoggingSetUp.class);
+
+        new StyxServerComponents.Builder()
+                .styxConfig(new StyxConfig())
+                .loggingSetUp(loggingSetUp)
+                .build();
+
+        verify(loggingSetUp).setUp(any(Environment.class));
+    }
+
+    @Test
+    public void loadsPlugins() {
+        Chain chain = mock(Chain.class);
+
+        List<NamedPlugin> pluginsForBuilder = ImmutableList.of(
+                namedPlugin("plugin1", stubPlugin("MyResponse1")),
+                namedPlugin("plugin2", stubPlugin("MyResponse2")));
+
+        StyxServerComponents components = new StyxServerComponents.Builder()
+                .styxConfig(new StyxConfig())
+                .plugins(env -> pluginsForBuilder)
+                .build();
+
+        List<NamedPlugin> plugins = ImmutableList.copyOf(components.plugins());
```

Comment:
Is there any good reason why the `components.plugins()` needs to return an `Iterable` instead of `List`? Now each consumer must always convert it to a list before it is any good.

Reply:
I will look into this.